src/common/io/filter/chunk.c

```diff
@@ -0,0 +1,172 @@
+/***********************************************************************************************************************************
```

Comment:
I look forward to seeing the overall description of chunking and its purpose.  Question: can it tie into the compression/encryption code as well?  With chunking, it becomes possible to append to a file which contains encrypted/compressed/signed data. It seems like a generally useful idea.

Reply:
(FYI - I'm assuming each chunk is individually encrypted/compressed or signed)

Reply:
(I'll update my comments as I read through ... still digesting it.)

Reply:
> I look forward to seeing the overall description of chunking and its purpose.

There is a brief description in the header: `Split data up into chunks so it can be written without knowing the eventual length of the data.` Basically the same idea as HTTP chunked encoding. We don't know how much data we are going to write up front, but this allows us to write as we go and the reader of the data can reassemble the chunks when reading sequentially.

> Question: can it tie into the compression/encryption code as well?

You can add this filter after cmp/enc and the output will be chunked.

> With chunking, it becomes possible to append to a file which contains encrypted/compressed/signed data. It seems like a generally useful idea.

Possibly, but appending to files in object stores is generally not allowed so not something we could take advantage of.